internal/api/middleware/oapi.go

```diff
@@ -0,0 +1,93 @@
+package middleware
+
+// based on https://github.com/deepmap/oapi-codegen/blob/9dc8b8d293a991614ea12447bd6507bfadf38304/pkg/chi-middleware/oapi_validate.go
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/openapi3filter"
+	"github.com/getkin/kin-openapi/routers"
+	"github.com/getkin/kin-openapi/routers/gorillamux"
+)
+
+// ErrorHandler is called when there is an error in validation
+type ErrorHandler func(w http.ResponseWriter, statusCode int, message string)
+
+// Options to customize request validation, openapi3filter specified options will be passed through.
+type Options struct {
+	Options openapi3filter.Options
+}
+
+// OapiRequestValidator Creates middleware to validate request by swagger spec.
+// This middleware is good for net/http either since go-chi is 100% compatible with net/http.
+func OapiRequestValidator(swagger *openapi3.T, handler ErrorHandler) func(next http.Handler) http.Handler {
+	return OapiRequestValidatorWithOptions(swagger, handler, nil)
+}
+
+// OapiRequestValidatorWithOptions Creates middleware to validate request by swagger spec.
+// This middleware is good for net/http either since go-chi is 100% compatible with net/http.
+func OapiRequestValidatorWithOptions(swagger *openapi3.T, handler ErrorHandler, options *Options) func(next http.Handler) http.Handler {
+	router, err := gorillamux.NewRouter(swagger)
+	if err != nil {
+		panic(err)
+	}
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			// validate request
+			if statusCode, err := validateRequest(r, router, options); err != nil {
+				handler(w, statusCode, err.Error())
```

Comment:
This is basically the one line that differs from the [original upstream](https://github.com/deepmap/oapi-codegen/blob/9dc8b8d293a991614ea12447bd6507bfadf38304/pkg/chi-middleware/oapi_validate.go#L42). The function signatures also have `handler ErrorHandler` passed in to them to make it so that this validation middleware can respond with our standard error handling helper (the `sendError` call that makes proper JSON messages) rather than returning a plain text error. I'll look into extending the `Options` values and submitting a PR upstream so we can get rid of this, but until then, don't think it's a big deal to just lift this here.

Reply:
BTW, just opened [this upstream PR](https://github.com/deepmap/oapi-codegen/pull/557), so if that gets merged any time soon we can just drop this additional middleware and just use the upstream library code.